extended/src/main/java/apoc/vectordb/VectorDb.java

```diff
@@ -0,0 +1,234 @@
+                                                                   Transaction tx,
+                                                                   Function<Object, Stream<Map>> objectMapper) throws Exception {
+        List<String> fields = procedureCallContext.outputFields().toList();
+
+        boolean hasVector = fields.contains("vector") && conf.isAllResults();
+        boolean hasMetadata = fields.contains("metadata");
+        Stream<Object> resultStream = executeRequest(conf.getApiConfig(), urlAccessChecker);
+
+        VectorMappingConfig mapping = conf.getMapping();
+
+        return resultStream
+                .flatMap(objectMapper)
+                .map(m -> getEmbeddingResult(conf, db, tx, hasVector, hasMetadata, mapping, m));
+    }
+
+    public static EmbeddingResult getEmbeddingResult(VectorEmbeddingConfig conf, GraphDatabaseService db, Transaction tx, boolean hasEmbedding, boolean hasMetadata, VectorMappingConfig mapping, Map m) {
+        Object id = conf.isAllResults() ? m.get(conf.getIdKey()) : null;
+        List<Double> embedding = hasEmbedding ? (List<Double>) m.get(conf.getVectorKey()) : null;
+        Map<String, Object> metadata = hasMetadata ? (Map<String, Object>) m.get(conf.getMetadataKey()) : null;
+        // in case of get operation, e.g. http://localhost:52798/collections/{coll_name}/points with Qdrant db,
+        // score is not present
+        Double score = Util.toDouble(m.get(conf.getScoreKey()));
+        String text = conf.isAllResults() ? (String) m.get(conf.getTextKey()) : null;
+
+        Entity entity = handleMapping(tx, db, mapping, metadata, embedding);
+        if (entity != null) entity = Util.rebind(tx, entity);
+        return new EmbeddingResult(id, score, embedding, metadata, text, entity);
+    }
+
+    private static Entity handleMapping(Transaction tx, GraphDatabaseService db, VectorMappingConfig mapping, Map<String, Object> metadata, List<Double> embedding) {
+        if (mapping.getProp() == null) {
+            return null;
+        }
+        if (MapUtils.isEmpty(metadata)) {
+            throw new RuntimeException("To use mapping config, the metadata should not be empty. Make sure you execute `YIELD metadata` on the procedure");
+        }
+        Map<String, Object> metaProps = new HashMap<>(metadata);
+        if (mapping.getLabel() != null) {
+            return handleMappingNode(tx, db, mapping, metaProps, embedding);
+        } else if (mapping.getType() != null) {
+            return handleMappingRel(tx, db, mapping, metaProps, embedding);
+        } else {
+            throw new RuntimeException("Mapping conf has to contain either label or type key");
+        }
+    }
+
+    private static Entity handleMappingNode(Transaction tx, GraphDatabaseService db, VectorMappingConfig mapping, Map<String, Object> metaProps, List<Double> embedding) {
+        String query = "CREATE CONSTRAINT IF NOT EXISTS FOR (n:%s) REQUIRE n.%s IS UNIQUE"
```

Comment:
did you test that you can run both the constraint as well as the data creation operation in the same tx? 

shouldn't we leave that to the user to create the constraint, otherwise it would do it for every entity